kafka-connector/src/com/axonivy/connector/kafka/KafkaStartEventBean.java

```diff
@@ -0,0 +1,336 @@
+
+		private KafkaConsumerRunnable(Properties properties, KafkaTopicConsumerSupplier<?, ?> topicConsumerSupplier, Duration pollTimeout) {
+			this.properties = properties;
+			this.topicConsumerSupplier = topicConsumerSupplier;
+			this.pollTimeout = pollTimeout;
+		}
+
+		@Override
+		public void run() {
+			try {
+				synchronous = isSynchronous();
+				log().debug("Handle Kafka messages synchronously: {0}", synchronous);
+
+				consumer = topicConsumerSupplier.apply(properties, getTopicPattern());
+				log().debug("Got consumer {0} which subscribed to Kafka topic pattern: {1}", consumer, getTopicPattern());
+
+				while(!consumerExecutor.isShutdown()) {
+					try {
+						ConsumerRecords<?, ?> records = consumer.poll(pollTimeout);
+						for (ConsumerRecord<?, ?> record : records) {
+							log().debug("Received record, topic: {0} offset:{1} key: {2} val: {3} record: {4}",
+									record.topic(), record.offset(), record.key(), record.value(), record);
+							KafkaWorkerRunnable kafkaWorkerRunnable = new KafkaWorkerRunnable(workerExecutor, record, consumer);
+							if(synchronous) {
+								kafkaWorkerRunnable.run();
+							}
+							else {
+								workerExecutor.execute(kafkaWorkerRunnable);
+							}
+						}
+					} catch (WakeupException e) {
+						log().info("Consumer was woken up. This is ok, when there is a shutdown.");
+					}
+				}
+			}
+			catch(Exception e) {
+				log().error("Exception while listening for topic, closing consumer.", e);
+			}
+			finally {
+				if(consumer != null) {
+					consumer.close();
+				}
+			}
+		}
+
+		public void wakeup() {
+			if(consumer != null) {
+				consumer.wakeup();
+			}
+		}
+	}
+
+	/**
+	 * A {@link ThreadFactory} which gives threads a configurable name.
+	 */
+	protected static class NamingThreadFactory implements ThreadFactory {
+
+		private String pattern;
+		private static ThreadFactory defaultThreadFactory;
+		private static AtomicInteger threadCount = new AtomicInteger(0);
+
+		protected NamingThreadFactory(String pattern) {
+			defaultThreadFactory = Executors.defaultThreadFactory();
+			this.pattern = pattern;
+		}
+
+		@Override
+		public Thread newThread(Runnable runnable) {
+			Thread thread = defaultThreadFactory.newThread(runnable);
+			thread.setName(String.format(pattern, threadCount.incrementAndGet()));
+			return thread;
+		}
+	}
+
+	/**
+	 * The editor to configure a {@link KafkaStartEventBean}.
+	 */
+	public static class Editor extends UiEditorExtension {
+		private IUiFieldEditor topicPatternField;
+		private IUiFieldEditor synchronousField;
+		private IUiFieldEditor kafkaConfigurationNameField;
+
+		@Override
+		public void initUiFields(ExtensionUiBuilder ui) {
+			ui.label("Topic Pattern:").create();
+			topicPatternField = ui.textField().create();
+
+			ui.label("Synchronous:").create();
+			synchronousField = ui.textField().create();
+
+			ui.label("Configuration Base:").create();
+			kafkaConfigurationNameField = ui.textField().create();

```

Comment:
good, that our new UI editors are getting used.
So we will gladly preserve it and support it with the next-gen inscription view :)